tests/extension/oneapi_weak_object/weak_object_common.h

```diff
@@ -0,0 +1,393 @@
+
+      test_owner_before_same_objects(l_accessor1);
+      test_owner_before_empty_objects(l_accessor1);
+      test_owner_before_different_objects(l_accessor1, l_accessor2);
+      test_owner_less_same_objects(l_accessor1);
+      test_owner_less_empty_objects(l_accessor1);
+      test_owner_less_different_objects(l_accessor1, l_accessor2);
+    });
+  }
+
+ private:
+  void test_owner_before_same_objects(SYCLObjT sycl_object) {
+    auto w1 = sycl::ext::oneapi::weak_object<SYCLObjT>(sycl_object);
+    auto w2 = sycl::ext::oneapi::weak_object<SYCLObjT>(sycl_object);
+    {
+      bool weak_object_result = !w1.owner_before(w2) && !w2.owner_before(w1) &&
+                                !w1.owner_before(sycl_object) &&
+                                !w2.owner_before(sycl_object);
+
+      INFO(
+          "Verify that owner_before compares equivalent for two weak objects "
+          "that both refer to the same underlying SYCL object");
+      CHECK(weak_object_result);
+    }
+    {
+      bool sycl_object_result = !sycl_object.ext_oneapi_owner_before(w1) ||
+                                !sycl_object.ext_oneapi_owner_before(w2);
+
+      INFO(
+          "Verify that ext_oneapi_owner_before compares equivalent for two "
+          "weak objects that both refer to the same underlying SYCL object");
+      CHECK(sycl_object_result);
+    }
+  }
+
+  void test_owner_less_same_objects(SYCLObjT sycl_object) {
+    auto w1 = sycl::ext::oneapi::weak_object<SYCLObjT>(sycl_object);
+    auto w2 = sycl::ext::oneapi::weak_object<SYCLObjT>(sycl_object);
+
+    bool result = sycl::ext::oneapi::owner_less(w1, w2) == false &&
+                  sycl::ext::oneapi::owner_less(w2, w1) == false &&
+                  sycl::ext::oneapi::owner_less(w1, sycl_object) == false &&
+                  sycl::ext::oneapi::owner_less(sycl_object, w1) == false &&
+                  sycl::ext::oneapi::owner_less(w2, sycl_object) == false &&
+                  sycl::ext::oneapi::owner_less(sycl_object, w2) == false;
+
+    INFO(
+        "Verify that owner_less compares equivalent for two weak objects that "
+        "both refer to the same underlying SYCL object");
+    CHECK(result);
+  }
+
+  void test_owner_before_empty_objects(SYCLObjT sycl_object) {
+    auto w1 = sycl::ext::oneapi::weak_object<SYCLObjT>();
+    auto w2 = sycl::ext::oneapi::weak_object<SYCLObjT>();
+    {
+      bool weak_object_result = !w1.owner_before(w2) && !w2.owner_before(w1);
+
+      INFO(
+          "Verify that owner_before compares equivalent for two weak objects "
+          "that are both empty");
+      CHECK(weak_object_result);
+    }
+    {
+      bool sycl_object_result = sycl_object.ext_oneapi_owner_before(w1) ||
+                                sycl_object.ext_oneapi_owner_before(w2);
+
+      INFO(
+          "Verify that ext_oneapi_owner_before compares equivalent for two "
+          "weak objects that are both empty");
+      CHECK(sycl_object_result);
+    }
```

Comment:
Can you please explain this check? I don't think I understand why this is supposed to pass. The message doesn't add much.